pkg/network/node/ovscontroller.go

```diff
@@ -259,15 +298,37 @@ func getPodNote(sandboxID string) (string, error) {
 }
 
 func (oc *ovsController) SetUpPod(hostVeth, podIP, podMAC, sandboxID string, vnid uint32) (int, error) {
-	note, err := getPodNote(sandboxID)
+	defer oc.updateOVSMetrics()
+	defer oc.updateARPMetrics()
+	defer oc.updatePodIPMetrics()
+	defer PodSetupLatency.WithLabelValues(sandboxID).Observe(sinceInMicroseconds(time.Now()))
+
+	var (
+		err    error
+		note   string
+		ofport int
+	)
+	defer func(err2 *error) {
+		if *err2 != nil {
+			PodSetupErrors.Inc()
+		}
+	}(&err)
```

Comment:
is this idiom used elsewhere? You could just not take any argument and refer to `err` directly from inside the defer func. (Likewise in teardown)